### ISSUE TITLE
Missing delete button on ClrSearch input

### ISSUE THREAD

**Issue by furti** [porscheinformatik/clarity-addons#566]
When the value of a the field is change programatically via the "ngModel" binding, the delte button of the component is not displayed.

The component only listens to the "input" event of the input field. But this is not triggered in this case. Only on user interaction the event is triggered.

I used a own directive as workaround that gets the NgControl injected and subscribes to the valueChanges.

Maybe this could be added in addition to the input event listener. Or maybe it could replace the listener completely.

```ts
private listener: () => void;
    private subscription: Subscription;

    constructor(private renderer: Renderer2, private inputEl: ElementRef, private ngControl: NgControl) {}

    ngOnInit() {
        this.subscription = this.ngControl.valueChanges.subscribe(value => {
            if (value) {
                this.renderer.addClass(this.inputEl.nativeElement.parentNode, "has-value");
            }

            window.setTimeout(() => this.unsubscribe(), 1);
        });
    }

    ngOnDestroy() {
        if (this.listener) {
            this.listener();

            delete this.listener;
        }

        this.unsubscribe();
    }

    private unsubscribe() {
        if (this.subscription) {
            this.subscription.unsubscribe();

            delete this.subscription;
        }
    }
```

**Comment by Wallabeng** [comment]
I will create a defect for this to fix it permanently in the library

**Comment by Wallabeng** [comment]
Fixed by #570
